build error with mxml version 4

TL;DR missing include, possibly breaking API changes in libmxml

Building against `>=dev-libs/mxml-4` on Gentoo fails: https://bugs.gentoo.org/935021

The build fails due to a missing include:

```
x86_64-pc-linux-gnu-gcc -Wall -c -O2 -march=x86-64 -pipe -pipe -frecord-gcc-switches -fno-diagnostics-color -fmessage-length=0 -O2 -DNDEBUG -D_FILE_OFFSET_BITS=64 -DFUSE_USE_VERSION=25 fuse-ts.c -o fuse-ts.o
fuse-ts-kdenlive.c:8:10: fatal error: mxml.h: No such file or directory
    8 | #include <mxml.h>
      |          ^~~~~~~~
compilation terminated.
fuse-ts-shotcut.c:8:10: fatal error: mxml.h: No such file or directory
    8 | #include <mxml.h>
      |          ^~~~~~~~
compilation terminated.
```

This is caused by a missing `-I/usr/include/libmxml4`, since the `mxml.h` file seems to have moved from `/usr/include` to `/usr/include/libmxml4` in version 4.
I suspect this change might have been made due to some breaking changes in the library.

As suspected, there are probably some incompatible API changes:

```
❯ LANG=C make
gcc -Wall -g -Wpedantic -c  -O2 -DNDEBUG -D_FILE_OFFSET_BITS=64 -DFUSE_USE_VERSION=25 -I/usr/include/libmxml4 fuse-ts-debug.c -o fuse-ts-debug.o
gcc -Wall -g -Wpedantic -c  -O2 -DNDEBUG -D_FILE_OFFSET_BITS=64 -DFUSE_USE_VERSION=25 -I/usr/include/libmxml4 fuse-ts-tools.c -o fuse-ts-tools.o
gcc -Wall -g -Wpedantic -c  -O2 -DNDEBUG -D_FILE_OFFSET_BITS=64 -DFUSE_USE_VERSION=25 -I/usr/include/libmxml4 fuse-ts-opts.c -o fuse-ts-opts.o
gcc -Wall -g -Wpedantic -c  -O2 -DNDEBUG -D_FILE_OFFSET_BITS=64 -DFUSE_USE_VERSION=25 -I/usr/include/libmxml4 fuse-ts-filelist.c -o fuse-ts-filelist.o
gcc -Wall -g -Wpedantic -c  -O2 -DNDEBUG -D_FILE_OFFSET_BITS=64 -DFUSE_USE_VERSION=25 -I/usr/include/libmxml4 fuse-ts-kdenlive.c -o fuse-ts-kdenlive.o
fuse-ts-kdenlive.c: In function 'find_cutmarks_in_kdenlive_project_file':
fuse-ts-kdenlive.c:210:40: error: passing argument 2 of 'mxmlLoadString' from incompatible pointer type [-Wincompatible-pointer-types]
  210 |         xmldoc = mxmlLoadString (NULL, temp, NULL);
      |                                        ^~~~
      |                                        |
      |                                        char *
In file included from fuse-ts-kdenlive.c:9:
/usr/include/libmxml4/mxml.h:182:75: note: expected 'mxml_options_t *' {aka 'struct _mxml_options_s *'} but argument is of type 'char *'
  182 | extern mxml_node_t      *mxmlLoadString(mxml_node_t *top, mxml_options_t *options, const char *s);
      |                                                           ~~~~~~~~~~~~~~~~^~~~~~~
In file included from fuse-ts-kdenlive.c:11:
fuse-ts.h:21:26: error: 'MXML_DESCEND' undeclared (first use in this function); did you mean 'MXML_DESCEND_ALL'?
   21 | #define MXML_DESCEND_ALL MXML_DESCEND
      |                          ^~~~~~~~~~~~
fuse-ts-kdenlive.c:217:80: note: in expansion of macro 'MXML_DESCEND_ALL'
  217 |         node = mxmlFindElement (xmldoc, xmldoc, "playlist", "id", "playlist0", MXML_DESCEND_ALL);
      |                                                                                ^~~~~~~~~~~~~~~~
fuse-ts.h:21:26: note: each undeclared identifier is reported only once for each function it appears in
   21 | #define MXML_DESCEND_ALL MXML_DESCEND
      |                          ^~~~~~~~~~~~
fuse-ts-kdenlive.c:217:80: note: in expansion of macro 'MXML_DESCEND_ALL'
  217 |         node = mxmlFindElement (xmldoc, xmldoc, "playlist", "id", "playlist0", MXML_DESCEND_ALL);
      |                                                                                ^~~~~~~~~~~~~~~~
fuse-ts-kdenlive.c: At top level:
fuse-ts-kdenlive.c:406:1: warning: string length '5187' is greater than the length '4095' ISO C99 compilers are required to support [-Woverlength-strings]
  406 | ;
      | ^
make: *** [Makefile:24: fuse-ts-kdenlive.o] Error 1
```

Unfortunately not. I can get it to work with a little modification to the source code (probably just a typo):

```diff
diff --git i/fuse-ts-xml.h w/fuse-ts-xml.h
index eb20cf5..2a49ce6 100644
--- i/fuse-ts-xml.h
+++ w/fuse-ts-xml.h
@@ -1,5 +1,5 @@
 
-#if MXML_VERSION_MAJOR < 4
+#if MXML_MAJOR_VERSION < 4
 #define MXML_DESCEND_ALL MXML_DESCEND
 #define XMLLOAD(x) mxmlLoadString (NULL, x, NULL);
 #else
```

However, link still fails because version 4 names its library `libmxml4.so`.
They provide a pkg-config file `libmxml4.pc`, though,
so this could make discovery of the compile and link flags a little easier.

```
❯ LANG=C make
gcc -Wall -g -Wpedantic -c  -O2 -DNDEBUG -D_FILE_OFFSET_BITS=64 -DFUSE_USE_VERSION=25 -I/usr/include/libmxml4 fuse-ts-debug.c -o fuse-ts-debug.o
gcc -Wall -g -Wpedantic -c  -O2 -DNDEBUG -D_FILE_OFFSET_BITS=64 -DFUSE_USE_VERSION=25 -I/usr/include/libmxml4 fuse-ts-tools.c -o fuse-ts-tools.o
gcc -Wall -g -Wpedantic -c  -O2 -DNDEBUG -D_FILE_OFFSET_BITS=64 -DFUSE_USE_VERSION=25 -I/usr/include/libmxml4 fuse-ts-opts.c -o fuse-ts-opts.o
gcc -Wall -g -Wpedantic -c  -O2 -DNDEBUG -D_FILE_OFFSET_BITS=64 -DFUSE_USE_VERSION=25 -I/usr/include/libmxml4 fuse-ts-filelist.c -o fuse-ts-filelist.o
gcc -Wall -g -Wpedantic -c  -O2 -DNDEBUG -D_FILE_OFFSET_BITS=64 -DFUSE_USE_VERSION=25 -I/usr/include/libmxml4 fuse-ts-kdenlive.c -o fuse-ts-kdenlive.o
fuse-ts-kdenlive.c:407:1: warning: string length '5187' is greater than the length '4095' ISO C99 compilers are required to support [-Woverlength-strings]
  407 | ;
      | ^
gcc -Wall -g -Wpedantic -c  -O2 -DNDEBUG -D_FILE_OFFSET_BITS=64 -DFUSE_USE_VERSION=25 -I/usr/include/libmxml4 fuse-ts-smoothsort.c -o fuse-ts-smoothsort.o
gcc -Wall -g -Wpedantic -c  -O2 -DNDEBUG -D_FILE_OFFSET_BITS=64 -DFUSE_USE_VERSION=25 -I/usr/include/libmxml4 fuse-ts-knowledge.c -o fuse-ts-knowledge.o
gcc -Wall -g -Wpedantic -c  -O2 -DNDEBUG -D_FILE_OFFSET_BITS=64 -DFUSE_USE_VERSION=25 -I/usr/include/libmxml4 fuse-ts-shotcut.c -o fuse-ts-shotcut.o
gcc -Wall -g -Wpedantic -c  -O2 -DNDEBUG -D_FILE_OFFSET_BITS=64 -DFUSE_USE_VERSION=25 -I/usr/include/libmxml4 fuse-ts-filebuffer.c -o fuse-ts-filebuffer.o
gcc -Wall -g -Wpedantic -c  -O2 -DNDEBUG -D_FILE_OFFSET_BITS=64 -DFUSE_USE_VERSION=25 -I/usr/include/libmxml4 fuse-ts.c -o fuse-ts.o
gcc fuse-ts-debug.o fuse-ts-tools.o fuse-ts-opts.o fuse-ts-filelist.o fuse-ts-kdenlive.o fuse-ts-smoothsort.o fuse-ts-knowledge.o fuse-ts-shotcut.o fuse-ts-filebuffer.o fuse-ts.o -Wall -g -Wpedantic -lmxml -lfuse  -O2 -DNDEBUG  -o fuse-ts
/usr/lib/gcc/x86_64-pc-linux-gnu/14/../../../../x86_64-pc-linux-gnu/bin/ld: cannot find -lmxml: No such file or directory
collect2: error: ld returned 1 exit status
make: *** [Makefile:27: fuse-ts] Error 1
```

Unfortunately, we missed something :')

https://github.com/a-tze/fuse-ts/blob/c1d42ee2dc9b6aba67348b4f386ecd91fb667e35/Makefile#L8

https://github.com/a-tze/fuse-ts/blob/c1d42ee2dc9b6aba67348b4f386ecd91fb667e35/Makefile#L12

You probably mean the same variable with `MXML_LIBS` and `MXML_LFLAGS`, but they have different names and so the value is empty when defining `LFLAGS`. Also, there is still an explicit `-lmxml`, but the version 4 library is `-lmxml4` (which is defined in `MXML_LIBS` by pkgconfig). I'll quickly open a PR for the fix.

This is what happens with **only** mxml4 on the system:

```
❯ LANG=C make
gcc -Wall -g -Wpedantic -c  -O2 -DNDEBUG -D_FILE_OFFSET_BITS=64 -DFUSE_USE_VERSION=25 -I/usr/include/libmxml4 fuse-ts-debug.c -o fuse-ts-debug.o
gcc -Wall -g -Wpedantic -c  -O2 -DNDEBUG -D_FILE_OFFSET_BITS=64 -DFUSE_USE_VERSION=25 -I/usr/include/libmxml4 fuse-ts-tools.c -o fuse-ts-tools.o
gcc -Wall -g -Wpedantic -c  -O2 -DNDEBUG -D_FILE_OFFSET_BITS=64 -DFUSE_USE_VERSION=25 -I/usr/include/libmxml4 fuse-ts-opts.c -o fuse-ts-opts.o
gcc -Wall -g -Wpedantic -c  -O2 -DNDEBUG -D_FILE_OFFSET_BITS=64 -DFUSE_USE_VERSION=25 -I/usr/include/libmxml4 fuse-ts-filelist.c -o fuse-ts-filelist.o
gcc -Wall -g -Wpedantic -c  -O2 -DNDEBUG -D_FILE_OFFSET_BITS=64 -DFUSE_USE_VERSION=25 -I/usr/include/libmxml4 fuse-ts-kdenlive.c -o fuse-ts-kdenlive.o
fuse-ts-kdenlive.c:407:1: warning: string length '5187' is greater than the length '4095' ISO C99 compilers are required to support [-Woverlength-strings]
  407 | ;
      | ^
gcc -Wall -g -Wpedantic -c  -O2 -DNDEBUG -D_FILE_OFFSET_BITS=64 -DFUSE_USE_VERSION=25 -I/usr/include/libmxml4 fuse-ts-smoothsort.c -o fuse-ts-smoothsort.o
gcc -Wall -g -Wpedantic -c  -O2 -DNDEBUG -D_FILE_OFFSET_BITS=64 -DFUSE_USE_VERSION=25 -I/usr/include/libmxml4 fuse-ts-knowledge.c -o fuse-ts-knowledge.o
gcc -Wall -g -Wpedantic -c  -O2 -DNDEBUG -D_FILE_OFFSET_BITS=64 -DFUSE_USE_VERSION=25 -I/usr/include/libmxml4 fuse-ts-shotcut.c -o fuse-ts-shotcut.o
gcc -Wall -g -Wpedantic -c  -O2 -DNDEBUG -D_FILE_OFFSET_BITS=64 -DFUSE_USE_VERSION=25 -I/usr/include/libmxml4 fuse-ts-filebuffer.c -o fuse-ts-filebuffer.o
gcc -Wall -g -Wpedantic -c  -O2 -DNDEBUG -D_FILE_OFFSET_BITS=64 -DFUSE_USE_VERSION=25 -I/usr/include/libmxml4 fuse-ts.c -o fuse-ts.o
gcc fuse-ts-debug.o fuse-ts-tools.o fuse-ts-opts.o fuse-ts-filelist.o fuse-ts-kdenlive.o fuse-ts-smoothsort.o fuse-ts-knowledge.o fuse-ts-shotcut.o fuse-ts-filebuffer.o fuse-ts.o -Wall -g -Wpedantic -lmxml -lfuse  -O2 -DNDEBUG   -o fuse-ts
/usr/lib/gcc/x86_64-pc-linux-gnu/14/../../../../x86_64-pc-linux-gnu/bin/ld: cannot find -lmxml: No such file or directory
collect2: error: ld returned 1 exit status
make: *** [Makefile:29: fuse-ts] Error 1
```

With both versions installed in parallel (notice that it includes `/usr/include/mxml4` in the include path, but links only against `-lmxml` instead of `-lmxml4`):

```
❯ LANG=C make
gcc -Wall -g -Wpedantic -c  -O2 -DNDEBUG -D_FILE_OFFSET_BITS=64 -DFUSE_USE_VERSION=25 -I/usr/include/libmxml4 fuse-ts-debug.c -o fuse-ts-debug.o
gcc -Wall -g -Wpedantic -c  -O2 -DNDEBUG -D_FILE_OFFSET_BITS=64 -DFUSE_USE_VERSION=25 -I/usr/include/libmxml4 fuse-ts-tools.c -o fuse-ts-tools.o
gcc -Wall -g -Wpedantic -c  -O2 -DNDEBUG -D_FILE_OFFSET_BITS=64 -DFUSE_USE_VERSION=25 -I/usr/include/libmxml4 fuse-ts-opts.c -o fuse-ts-opts.o
gcc -Wall -g -Wpedantic -c  -O2 -DNDEBUG -D_FILE_OFFSET_BITS=64 -DFUSE_USE_VERSION=25 -I/usr/include/libmxml4 fuse-ts-filelist.c -o fuse-ts-filelist.o
gcc -Wall -g -Wpedantic -c  -O2 -DNDEBUG -D_FILE_OFFSET_BITS=64 -DFUSE_USE_VERSION=25 -I/usr/include/libmxml4 fuse-ts-kdenlive.c -o fuse-ts-kdenlive.o
fuse-ts-kdenlive.c:407:1: warning: string length '5187' is greater than the length '4095' ISO C99 compilers are required to support [-Woverlength-strings]
  407 | ;
      | ^
gcc -Wall -g -Wpedantic -c  -O2 -DNDEBUG -D_FILE_OFFSET_BITS=64 -DFUSE_USE_VERSION=25 -I/usr/include/libmxml4 fuse-ts-smoothsort.c -o fuse-ts-smoothsort.o
gcc -Wall -g -Wpedantic -c  -O2 -DNDEBUG -D_FILE_OFFSET_BITS=64 -DFUSE_USE_VERSION=25 -I/usr/include/libmxml4 fuse-ts-knowledge.c -o fuse-ts-knowledge.o
gcc -Wall -g -Wpedantic -c  -O2 -DNDEBUG -D_FILE_OFFSET_BITS=64 -DFUSE_USE_VERSION=25 -I/usr/include/libmxml4 fuse-ts-shotcut.c -o fuse-ts-shotcut.o
gcc -Wall -g -Wpedantic -c  -O2 -DNDEBUG -D_FILE_OFFSET_BITS=64 -DFUSE_USE_VERSION=25 -I/usr/include/libmxml4 fuse-ts-filebuffer.c -o fuse-ts-filebuffer.o
gcc -Wall -g -Wpedantic -c  -O2 -DNDEBUG -D_FILE_OFFSET_BITS=64 -DFUSE_USE_VERSION=25 -I/usr/include/libmxml4 fuse-ts.c -o fuse-ts.o
gcc fuse-ts-debug.o fuse-ts-tools.o fuse-ts-opts.o fuse-ts-filelist.o fuse-ts-kdenlive.o fuse-ts-smoothsort.o fuse-ts-knowledge.o fuse-ts-shotcut.o fuse-ts-filebuffer.o fuse-ts.o -Wall -g -Wpedantic -lmxml -lfuse  -O2 -DNDEBUG   -o fuse-ts
```

**only** mxml3 installed (everything works as it should here):

```
❯ LANG=C make
gcc -Wall -g -Wpedantic -c  -O2 -DNDEBUG -D_FILE_OFFSET_BITS=64 -DFUSE_USE_VERSION=25  fuse-ts-debug.c -o fuse-ts-debug.o
gcc -Wall -g -Wpedantic -c  -O2 -DNDEBUG -D_FILE_OFFSET_BITS=64 -DFUSE_USE_VERSION=25  fuse-ts-tools.c -o fuse-ts-tools.o
gcc -Wall -g -Wpedantic -c  -O2 -DNDEBUG -D_FILE_OFFSET_BITS=64 -DFUSE_USE_VERSION=25  fuse-ts-opts.c -o fuse-ts-opts.o
gcc -Wall -g -Wpedantic -c  -O2 -DNDEBUG -D_FILE_OFFSET_BITS=64 -DFUSE_USE_VERSION=25  fuse-ts-filelist.c -o fuse-ts-filelist.o
gcc -Wall -g -Wpedantic -c  -O2 -DNDEBUG -D_FILE_OFFSET_BITS=64 -DFUSE_USE_VERSION=25  fuse-ts-kdenlive.c -o fuse-ts-kdenlive.o
fuse-ts-kdenlive.c:407:1: warning: string length '5187' is greater than the length '4095' ISO C99 compilers are required to support [-Woverlength-strings]
  407 | ;
      | ^
gcc -Wall -g -Wpedantic -c  -O2 -DNDEBUG -D_FILE_OFFSET_BITS=64 -DFUSE_USE_VERSION=25  fuse-ts-smoothsort.c -o fuse-ts-smoothsort.o
gcc -Wall -g -Wpedantic -c  -O2 -DNDEBUG -D_FILE_OFFSET_BITS=64 -DFUSE_USE_VERSION=25  fuse-ts-knowledge.c -o fuse-ts-knowledge.o
gcc -Wall -g -Wpedantic -c  -O2 -DNDEBUG -D_FILE_OFFSET_BITS=64 -DFUSE_USE_VERSION=25  fuse-ts-shotcut.c -o fuse-ts-shotcut.o
gcc -Wall -g -Wpedantic -c  -O2 -DNDEBUG -D_FILE_OFFSET_BITS=64 -DFUSE_USE_VERSION=25  fuse-ts-filebuffer.c -o fuse-ts-filebuffer.o
gcc -Wall -g -Wpedantic -c  -O2 -DNDEBUG -D_FILE_OFFSET_BITS=64 -DFUSE_USE_VERSION=25  fuse-ts.c -o fuse-ts.o
gcc fuse-ts-debug.o fuse-ts-tools.o fuse-ts-opts.o fuse-ts-filelist.o fuse-ts-kdenlive.o fuse-ts-smoothsort.o fuse-ts-knowledge.o fuse-ts-shotcut.o fuse-ts-filebuffer.o fuse-ts.o -Wall -g -Wpedantic -lmxml -lfuse  -O2 -DNDEBUG   -o fuse-ts
```

--- a/Makefile
+++ b/Makefile
@@ -9,7 +9,7 @@ MXML_LIBS := $(shell pkg-config --libs mxml$(MXML_VERSION))
 DEBUGopts = -g -O0 -fno-inline-functions -DDEBUG
 NDEBUGopts = $(EXTRA_CFLAGS) -O2 -DNDEBUG
 CFLAGS = -Wall -g -Wpedantic -c $(DEBUG) -D_FILE_OFFSET_BITS=64 -DFUSE_USE_VERSION=25 $(MXML_CFLAGS)
-LFLAGS = -Wall -g -Wpedantic -lmxml -lfuse $(DEBUG) $(EXTRA_LFLAGS) $(MXML_LFLAGS)
+LFLAGS = -Wall -g -Wpedantic -lfuse $(DEBUG) $(EXTRA_LFLAGS) $(MXML_LIBS)
 CC = gcc
 DEBUG=$(NDEBUGopts)
 

